### DISCUSSION TITLE
Github Actions IP Ranges for Whitelist

### DISCUSSION THREAD

**Discussion by GoldFlsh** [community/community#26884] [Actions]
<p>Hello, we are trying to see if it’s possible to whitelist github actions IP Ranges to ensure that only github actions runners can assume a particular role in AWS. Utilizing the following action: <a href="https://github.com/aws-actions/configure-aws-credentials" rel="nofollow noopener">https://github.com/aws-actions/configure-aws-credentials</a></p>
<p>I’m a little confused by the description for github actions IP Ranges. Described here:<br>
<a href="https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#ip-addresses-of-github-hosted-runners" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#ip-addresses-of-github-hosted-runners</a></p>
<p>First question is that the docs says that the IP Ranges+ServiceTags JSON file is updated weekly. Does that mean the IP Ranges changes every week? Do we need to continuously update our whitelist in line with the values in the JSON? Or now that I’ve downloaded this file, the values in it should remain relatively static?</p>
<p>Second, it says that the IP Address ranges are the same as <code>Azure Data center</code> and uses an example of IP Address prefixes from <code>AzureCloud.eastus2</code> is <code>AzureCloud</code> synonymous with <code>Azure Data center</code> in this context? and we would just need to find all the IP Ranges in <code>AzureCloud.[eastus, eastus2, westus2...]</code>?</p>
<p>Third, I happened to be scrolling through the list and saw the service <code>ActionGroup</code> is that something different from Github Actions? Or can I use the address prefixes in these lists instead of whitelisting all of Microsoft Azure?</p>
<pre><code class="lang-auto">{
      "name": "ActionGroup.EastUS",
      "id": "ActionGroup.EastUS",
      "properties": {
        ...
      }
    },
</code></pre>
<pre><code class="lang-auto">{
      "name": "ActionGroup",
      "id": "ActionGroup",
      "properties": {
        ...
    },
</code></pre>

**Comment by GoldFlsh** [comment]
<p>I found a partial answer to my first question in the fine print:</p>
<blockquote>
<p>This file is updated weekly. New ranges appearing in the file will not be used in Azure for at least one week. Please download the new json file every week and perform the necessary changes at your site to correctly identify services running in Azure.</p>
</blockquote>
<p>This leads to a follow-up, Is it possible for IP Addresses to drop out of the list every week or is it entirely for adding new ranges. Based on the wording, it seems like it could just be that they may add a new IP every once and a while.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/goldflsh">@GoldFlsh</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<ol>
<li>Yes, it’s possible for IP address  to drop out of the list.</li>
<li>Please see <a href="https://github.com/andymckay/actions-ips" rel="nofollow noopener">https://github.com/andymckay/actions-ips</a> as an open source library to get and update the list of IPs from the list. We’ll try and keep that up to date as things change. Specifically <a href="https://github.com/andymckay/actions-ips/blob/262a439a34e535ae2c77f4cf49cbad270c456d95/ips.py#L25-L32" rel="nofollow noopener">https://github.com/andymckay/actions-ips/blob/262a439a34e535ae2c77f4cf49cbad270c456d95/ips.py#L25-L32</a> as the zones we use.</li>
<li>Please ignore AzureCloud.</li>
</ol>
<p>Thanks.</p>

**Comment by GoldFlsh** [comment]
<p>The list is too large to use for my use case, since AWS trust policies have a size limit. But the answer addresses my main confusion around the language in the github docs, thank you :slight_smile:</p>

**Comment by ushuz** [comment]
<p>The whitelist is used to restrict access to protected resources, like private registry.  Whitelisting IP ranges of Azure’s regions basically allows entire Azure region to access said resources. This is not a viable practice. I hope GitHub could limit runner IPs to a fixed range.</p>

**Reply by gfl-akovalchuk** [reply]
5 years and still no solution :(

**Reply by melroy89** [reply]
Solution is here: https://docs.github.com/en/rest/meta/meta?apiVersion=2026-03-10#get-github-meta-information

**Comment by K-adarshCashfree** [comment]
Hello, 
We want to whitelist standard github-hosted runner IPs [github action IPs] into access policies of AWS resources. Since this list is huge and there is a character limit on policies, need to know how can we whitelist them.

**Reply by gfl-akovalchuk** [reply]
As far as I know there is no way to solve this issue using whitelist.
We started to use our own runner for deployment step.
